### PR TITLE
Some additional fixes

### PR DESCRIPTION
**Pull request by kimpaller**
1. fix: remove trailing characters in target path
2. support dmesg check for pluto
3.  retry JTAG connect attempt on failure 
4. jtag: fix logging issue

### CODE REVIEW

File: nebula/network.py [hunk 1]
```diff
@@ -246,12 +247,21 @@ def check_dmesg(self, error_on_warnings=False):
...
-        self.run_ssh_command("dmesg -l warn > " + tmp_filename_war)
-        self.run_ssh_command("dmesg -l err > " + tmp_filename_err)
-        self._dl_file(tmp_filename)
-        self._dl_file(tmp_filename_war)
-        self._dl_file(tmp_filename_err)
+        if self.board_name == 'pluto':
+            with open(tmp_filename_root, 'w') as outfile:
+                outfile.write(self.run_ssh_command("dmesg").stdout)
+            with open(tmp_filename_root + "_warn", 'w') as outfile:
+                outfile.write(self.run_ssh_command('dmesg -r | { grep "^.4" || true; }').stdout)
```
Comment (tagoylo) [inline nebula/network.py:254]: What does the ".4" in grep indicate?

File: nebula/network.py [hunk 2]
```diff
@@ -246,12 +247,21 @@ def check_dmesg(self, error_on_warnings=False):
...
+            with open(tmp_filename_root + "_err", 'w') as outfile:
+                outfile.write(self.run_ssh_command('dmesg -r | { grep "^.3" || true; }').stdout)
```
Comment (tagoylo) [inline nebula/network.py:256]: What does the ".3" in grep indicate?

Comment (tagoylo) [reply nebula/network.py:256]: Line 279 might not work with pluto's dmesg err.

File: nebula/network.py (hunk 1 above)
Comment (kimpaller) [reply nebula/network.py:254]: 4 denotes dmesg logs of level 'warning' while
3 denotes dmesg logs of level 'error'

File: nebula/network.py (hunk 2 above)
Comment (kimpaller) [reply nebula/network.py:256]: modfied code to retain the timestamp (remove strip()) ., with this mode., all dmesg logs even without the timestamp (like pluto dmesgs) will be processed without errors.,

**Review by tagoylo — APPROVED** [review]

**Pull request merged**